frontend/src/store/objectStore.ts

```diff
@@ -123,15 +129,24 @@ export const useObjectStore = defineStore('object', () => {
 
         let response = Array<COMSObject>();
         if (uniqueIds.length) {
+          // If metadata specified, search for objects with matching metadata
+          const headers: Record<string, string> = {};
+          if (metadata?.length) {
+            for (const meta of metadata) {
+              headers[`x-amz-meta-${meta.key}`] = meta.value;
+            }
+          }
+
           response = (await objectService.searchObjects({
             bucketId: params.bucketId ? [params.bucketId] : undefined,
             objectId: uniqueIds,
+            tagset: tagset ? tagset.reduce((acc, cur) => ({ ...acc, [cur.key]: cur.value }), {}) : undefined,
 
             // Added to allow deletion of objects before versioning implementation
             // TODO: Verify if needed after versioning implemented
             deleteMarker: false,
             latest: true
-          })).data;
+          }, headers)).data;
 
           // Remove old values matching search parameters
           const matches = (x: COMSObject) => (
```

Comment:
regarding the issue where the search results are doing an OR instead of the desired AND behavior, i wonder if we need to remove old values here